apple/Handlers/RNNativeViewHandler.mm

```diff
@@ -69,6 +76,18 @@ - (void)reset
   [_gestureHandler reset];
 }
 
+- (void)setStateIfScrollView:(UIGestureRecognizerState)state
+{
+  if ([self isScrollView]) {
+    self.state = state;
+  }
+}
+
+- (BOOL)isScrollView
+{
+  return [self.view isKindOfClass:[RCTScrollView class]];
```

Comment:
On the new architecture you need to check for `RCTScrollViewComponentView`, I think there's a similar check already in the same file so you can do it similarly.